### DISCUSSION TITLE
ESP8266 does not send UDP packet to Ubuntu server even though it has the correct return code

### DISCUSSION THREAD

**Discussion by mikronauts** [micropython/micropython#15780] [ESP8266]
I have a weird issue.

Ubuntu server 24.04, firewall configured to accept the udp message.

ESP8266 nodemcu board flashed with MicroPython v1.23.0 on 2024-06-02

Connects to my WiFi, says it sent the UDP datagram, however nothing is received by Ubuntu, no log of it being received.

If I send the datagram from another ubuntu box it works fine.

I will post the send and receive python code later, and will try esp-esp udp, as well as an esp32 sender, as well.

Any ideas?

Thanks,

Bill

**Comment by bidrohini** [comment]
Use Wireshark or tcpdump on your Ubuntu server to monitor incoming UDP traffic. This will help you verify if the packet is even reaching the network interface.
Example command: sudo tcpdump -i eth0 udp port <your_port_number>.

**Comment by mikronauts** [comment]
Thank you, that was going to be my next step, but I managed to solve the issue this morning

**Comment by mikronauts** [comment]
This morning I decided to try an ESP32 module.

I tried to update it to the latest MicroPython image, but I could not erase or program it initially.

Turns out Ubuntu added a udev rule that blocked chg340 serial adapters in favour of a keyboard dongle:

https://github.com/juliagoda/CH341SER/issues/18

next I updated my pytool, and flashed the new rom - and my UDP client worked on the ESP32

I went back and reflashed an ESP8266 with the new firmware, and now it works!
